File: solutions/2133. Check if Every Row and Column Contains All Numbers/2133.java
class Solution {
  public boolean checkValid(int[][] matrix) {
    final int n = matrix.length;

    for (int i = 0; i < n; ++i) {
      Set<Integer> row = new HashSet<>();
      Set<Integer> col = new HashSet<>();
      for (int j = 0; j < n; ++j) {
        row.add(matrix[i][j]);
        col.add(matrix[j][i]);
      }
      if (Math.min(row.size(), col.size()) < n)
        return false;
    }

    return true;
  }
}

File: solutions/2133. Check if Every Row and Column Contains All Numbers/2133.cpp
class Solution {
 public:
  bool checkValid(vector<vector<int>>& matrix) {
    const int n = matrix.size();

    for (int i = 0; i < n; ++i) {
      bitset<101> row;
      bitset<101> col;
      for (int j = 0; j < n; ++j) {
        row[matrix[i][j]] = true;
        col[matrix[j][i]] = true;
      }
      if (min(row.contains(), col.contains()) < n)
        return false;
    }

    return true;
  }
};
